client/src/sections/courses/LearningSection/AddLearnerModal.tsx

```diff
@@ -1,125 +1,184 @@
+  const { isEnrollee } = useSelector((state: RootState) => state.courseLearner);
 
-  const handleModalClose = (): void => {
-    setIsModalOpen(false);
+  const [mutateAsync] = useEnrollLearnerMutation();
+
+  const { data: trainee, refetch } = useGetLearnerQuery({
+    courseId: courseID,
+    isEnrolled: isEnrollee,
+    pageNumber: page,
+    searchQuery: search,
+  });
+  const enrollees = trainee?.learners.data;
+
+  const handleAddMaterialModal = (): void => {
+    setSelectedIds([]);
+    setSearch('');
+    setPage(1);
+    setIsModalOpen(!isModalOpen);
+    dispatch(toggleLearner());
+    if (!isEnrollee) {
+      void refetch();
+    }
   };
 
   const handleSearch = (search: string): void => {
-    alert(`You searched ${search}`);
+    setSearch(search);
+    setPage(1);
   };
 
-  const learnersPerPage = 8;
-  const totalPages = Math.ceil(learners.length / learnersPerPage);
+  const handleChangePageEvent = (page: number): void => {
+    setPage(page);
+  };
 
-  const lastIndex = currentPage * learnersPerPage;
-  const firstIndex = lastIndex - learnersPerPage;
-  const currentLearners = learners.slice(firstIndex, lastIndex);
+  const handleCheckboxChange = (id: number): void => {
+    if (selectedIds.includes(id)) {
+      setSelectedIds(selectedIds.filter((selectedId) => selectedId !== id));
+      setSelectedEnrollees(selectedEnrollees.filter((enrollee) => enrollee.trainee_id !== id));
+    } else {
+      setSelectedIds([...selectedIds, id]);
+      const selectedEnrollee = enrollees.find((enrollee: Trainee) => enrollee.trainee_id === id);
+      if (selectedEnrollee) {
+        setSelectedEnrollees([...selectedEnrollees, selectedEnrollee]);
+      }
+    }
+  };
 
-  const handleChangePageEvent = (page: number): void => {
-    setCurrentPage(page);
+  const handleAddLearners = async (): Promise<void> => {
+    const postData = {
+      trainee: selectedIds.join(','),
+    };
+    try {
+      await mutateAsync({
+        courseId: courseID,
+        postData,
+      });
+      dispatch(addTrainees(selectedEnrollees));
+      handleAddMaterialModal();
+    } catch (error) {
+      console.error('An error occurred while enrolling learners:', error);
+    }
   };
 
   return (
     <div>
       <Button
-            text="Add learner"
-            buttonClass="px-4 py-2 text-sm bg-white text-blue-500 border-2 border-red"
-            textColor="text-red"
-            onClick={() => { setIsModalOpen(true); } }
-          />
-    <Modal isOpen={isModalOpen} className="w-[521px]">
-      <div className="p-4">
-        <div className="flex justify-between mb-6">
-          <span className="text-gray-600 font-bold">Add Learner</span>
-          <div onClick={handleModalClose}>
-            <CloseIcon className="cursor-pointer w-[30px] h-[30px]" />
+        text="Add learner"
+        buttonClass="px-4 py-2 text-sm bg-white text-blue-500 border-2 border-red"
+        textColor="text-red"
+        onClick={handleAddMaterialModal}
+      />
```

Comment:
Can we move this button in the index file instead? 🙇‍♂️
Since you are rendering this component without condition in index, even though the modal is not open, its fetching not enrolled trainees when user visits the tab

That would be great if we can move button in index file, so we would conditionally render the component so that the logic inside it would get triggered only when its needed

But if it takes time, its okay to skip it for now